common/src/test/scala/uk/ac/wellcome/models/MergedSierraRecordTest.scala

```diff
@@ -7,236 +7,289 @@ import uk.ac.wellcome.utils.JsonUtil
+          id = "b111",
+          itemData = Map(record.id -> record)
+        )
 
-  it("should not perform a transformation without bibData") {
-    val mergedSierraRecord =
-      MergedSierraRecord(id = "000", maybeBibData = None)
+        val newerRecord = sierraItemRecord(
+          id = record.id,
+          title = "Old otters outside the oblong",
+          modifiedDate = "2000-01-01T01:01:01Z",
+          bibIds = List("i111")
+        )
+        val result = mergedSierraRecord.mergeItemRecord(newerRecord)
+        result shouldBe mergedSierraRecord
+      }
 
-    val transformedSierraRecord = mergedSierraRecord.transform
-    transformedSierraRecord.isSuccess shouldBe true
+      it("should support adding multiple items to a merged record") {
+        val record1 = sierraItemRecord(
+          id = "i111",
+          title = "Outside the orangutan opens an orange",
+          modifiedDate = "2001-01-01T01:01:01Z",
+          bibIds = List("b121")
+        )
+        val record2 = sierraItemRecord(
+          id = "i222",
+          title = "Twice the turtles took the turn",
+          modifiedDate = "2002-02-02T02:02:02Z",
+          bibIds = List("b121")
+        )
 
-    transformedSierraRecord.get shouldBe None
-  }
+        val mergedSierraRecord = MergedSierraRecord(id = "b121")
+        val result1 = mergedSierraRecord.mergeItemRecord(record1)
+        val result2 = result1.mergeItemRecord(record2)
+
+        result1.itemData(record1.id) shouldBe record1
+        result2.itemData(record2.id) shouldBe record2
+      }
+    }
 
-  it("should not perform a transformation, even if some itemData is present") {
-    val mergedSierraRecord = MergedSierraRecord(
-      id = "b111",
-      maybeBibData = None,
-      itemData = Map(
-        "i111" -> sierraItemRecord(
+    describe("when the merged record is unlinked from the item") {
+      ignore("should remove the item if it already exists") {
+        val unlinkedBibId = "222"
+        val record = sierraItemRecord(
           id = "i111",
-          title = "An incomplete invocation of items",
-          modifiedDate = "2001-01-01T01:01:01Z"
-        ))
-    )
-
-    val transformedSierraRecord = mergedSierraRecord.transform
-    transformedSierraRecord.isSuccess shouldBe true
-    transformedSierraRecord.get shouldBe None
+          title = "Only otters occupy the orange oval",
+          modifiedDate = "2001-01-01T01:01:01Z",
+          bibIds = List("i111"),
+          unlinkedBibIds = List(unlinkedBibId)
+        )
+        val mergedSierraRecord = MergedSierraRecord(id = unlinkedBibId,
+                                                    itemData =
+                                                      Map(record.id -> record))
+        mergedSierraRecord.unlinkItemRecord(record) shouldBe mergedSierraRecord
+          .copy(itemData = Map.empty)
+      }
+
+      ignore(
+        "should return None when merging an unlinked record which is already absent") {
```

Comment:
Although we’re ignoring these tests for now, do you want to update the descriptions (i.e. not *should return None*).

Reply:
👍